pkg/pillar/types/zedmanagertypes.go

```diff
@@ -276,28 +278,160 @@ type StorageStatus struct {
-	ss.CertificateChain = sc.CertificateChain
-	ss.ImageSignature = sc.ImageSignature
-	ss.SignatureKey = sc.SignatureKey
-	ss.ReadOnly = sc.ReadOnly
-	ss.Preserve = sc.Preserve
-	ss.Format = sc.Format
-	ss.Maxsizebytes = sc.Maxsizebytes
-	ss.Devtype = sc.Devtype
-	ss.Target = sc.Target
-	if ss.Format == "container" {
-		ss.IsContainer = true
+func (ssPtr *StorageStatus) UpdateFromStorageConfig(sc StorageConfig) {
+	ssPtr.DatastoreID = sc.DatastoreID
+	ssPtr.Name = sc.Name
+	ssPtr.NameIsURL = sc.NameIsURL
+	ssPtr.ImageSha256 = sc.ImageSha256
+	ssPtr.Size = sc.Size
+	ssPtr.CertificateChain = sc.CertificateChain
+	ssPtr.ImageSignature = sc.ImageSignature
+	ssPtr.SignatureKey = sc.SignatureKey
+	ssPtr.ReadOnly = sc.ReadOnly
+	ssPtr.Preserve = sc.Preserve
+	ssPtr.Format = sc.Format
+	ssPtr.Maxsizebytes = sc.Maxsizebytes
+	ssPtr.Devtype = sc.Devtype
+	ssPtr.Target = sc.Target
+	if ssPtr.Format == "container" {
+		ssPtr.IsContainer = true
 	}
 	return
 }
 
+// CheckForCerts checks certificate requirement/availability for a storage object
+// True, when there is no Certs or, the certificates are ready
+// False, Certificates are not ready or, there are some errors
+func (ssPtr *StorageStatus) CheckForCerts(safename string, checkCerts bool,
+	uuidStr string, certObjStatusPtr *CertObjStatus) bool {
+	if checkCerts && ssPtr.NeedsCerts() && ssPtr.GetCertCount() != 0 {
+		if !ssPtr.CheckCertsStatusForObject(uuidStr, certObjStatusPtr) {
+			log.Infof("MaybeAddVerifyImageConfig for %s, Certs are still not ready\n",
+				safename)
+			return false
+		}
+		if !ssPtr.CheckCertsForObject() {
+			log.Infof("MaybeAddVerifyImageConfig for %s, Certs are still not installed\n",
+				safename)
+			return false
+		}
+	}
+	return true
+}
+
+// NeedsCerts certificates are required for the Storage Object
+func (ssPtr *StorageStatus) NeedsCerts() bool {
+	if len(ssPtr.ImageSignature) != 0 {
+		return true
+	}
+	return false
+}
+
+// GetCertCount returns the number of certificates for the Storage Object
+func (ssPtr *StorageStatus) GetCertCount() int {
+	cidx := 0
+	if ssPtr.SignatureKey != "" {
+		cidx++
+	}
+	if len(ssPtr.CertificateChain) != 0 {
+		for _, certURL := range ssPtr.CertificateChain {
+			if certURL == "" {
+				log.Fatalf("Empty CertURL for %s\n", ssPtr.Name)
+			}
+			cidx++
+		}
+	}
+	return cidx
+}
+
+// CheckCertsStatusForObject checks certificates for installation status
+func (ssPtr *StorageStatus) CheckCertsStatusForObject(uuidStr string,
+	certObjStatusPtr *CertObjStatus) bool {
+
+	// certificates are still not ready, for processing
+	if certObjStatusPtr == nil {
+		log.Errorf("certObj Status is still not ready for %s\n", uuidStr)
+		return false
+	}
+
+	if ssPtr.SignatureKey != "" {
+		for _, certObj := range certObjStatusPtr.StorageStatusList {
+			if certObj.Name == ssPtr.SignatureKey {
+				if certObj.Error != "" {
+					errSrc := pubsub.TypeToName(VerifyImageStatus{})
+					ssPtr.SetErrorInfo(certObj.Error, certObj.ErrorTime, errSrc)
```

Comment:
A Check* function shouldn't have any side effects like modifying the object. Please make this function return bool, error and have setting of error info be done at the top caller.

Reply:
Moved it back to the calling routine.